madara/crates/client/settlement_client/src/eth/mod.rs

```diff
@@ -0,0 +1,862 @@
+    async fn get_last_event_block_number_works(eth_client: EthereumClient) {
+        let block_number = eth_client
+            .get_last_event_block_number()
+            .await
+            .expect("issue while getting the last block number with given event");
+        assert_eq!(block_number, L1_BLOCK_NUMBER, "block number with given event not matching");
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn get_last_verified_block_hash_works(eth_client: EthereumClient) {
+        let block_hash =
+            eth_client.get_last_verified_block_hash().await.expect("issue while getting the last verified block hash");
+        let expected = u256_to_felt(U256::from_str_radix(L2_BLOCK_HASH, 10).unwrap()).unwrap();
+        assert_eq!(block_hash, expected, "latest block hash not matching");
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn get_last_state_root_works(eth_client: EthereumClient) {
+        let state_root = eth_client.get_last_verified_state_root().await.expect("issue while getting the state root");
+        let expected = u256_to_felt(U256::from_str_radix(L2_STATE_ROOT, 10).unwrap()).unwrap();
+        assert_eq!(state_root, expected, "latest block state root not matching");
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn get_last_verified_block_number_works(eth_client: EthereumClient) {
+        let block_number = eth_client.get_last_verified_block_number().await.expect("issue");
+        assert_eq!(block_number, L2_BLOCK_NUMBER, "verified block number not matching");
+    }
+}
+
+#[cfg(test)]
+mod l1_messaging_tests {
+
+    use std::{sync::Arc, time::Duration};
+
+    use self::DummyContract::DummyContractInstance;
+    use crate::client::SettlementClientTrait;
+    use crate::eth::{EthereumClient, StarknetCoreContract};
+    use crate::messaging::{sync, L1toL2MessagingEventData};
+    use alloy::{
+        hex::FromHex,
+        node_bindings::{Anvil, AnvilInstance},
+        primitives::{Address, U256},
+        providers::{ProviderBuilder, RootProvider},
+        sol,
+        transports::http::{Client, Http},
+    };
+    use mc_db::DatabaseService;
+    use mc_mempool::{GasPriceProvider, L1DataProvider, Mempool, MempoolLimits};
+    use mp_chain_config::ChainConfig;
+    use mp_utils::service::ServiceContext;
+    use rstest::*;
+    use starknet_api::core::Nonce;
+    use starknet_types_core::felt::Felt;
+    use tracing_test::traced_test;
+    use url::Url;
+
+    struct TestRunner {
+        #[allow(dead_code)]
+        anvil: AnvilInstance, // Not used but needs to stay in scope otherwise it will be dropped
+        db_service: Arc<DatabaseService>,
+        dummy_contract: DummyContractInstance<Http<Client>, RootProvider<Http<Client>>>,
+        eth_client: EthereumClient,
+        mempool: Arc<Mempool>,
+    }
+
+    // LogMessageToL2 from https://etherscan.io/tx/0x21980d6674d33e50deee43c6c30ef3b439bd148249b4539ce37b7856ac46b843
+    // bytecode is compiled DummyContractBasicTestCase
```

Comment:
Would be good to mention the compiler/version or any other hints about reproducing this bytecode